libsolidity/formal/CHC.cpp

```diff
@@ -1458,51 +1462,28 @@ optional<string> CHC::generateCounterexample(CHCSolverInterface::CexGraph const&
 	vector<string> path;
 	string localState;
 
-	unsigned node = *rootId;
-	/// The first summary node seen in this loop represents the last transaction.
-	bool lastTxSeen = false;
-	while (_graph.edges.at(node).size() >= 1)
-	{
-		auto const& edges = _graph.edges.at(node);
-		solAssert(edges.size() <= 2, "");
-
-		unsigned summaryId = edges.at(0);
-		optional<unsigned> interfaceId;
-		if (edges.size() == 2)
-		{
-			interfaceId = edges.at(1);
-			if (!Predicate::predicate(_graph.nodes.at(summaryId).name)->isSummary())
-				swap(summaryId, *interfaceId);
-			auto interfacePredicate = Predicate::predicate(_graph.nodes.at(*interfaceId).name);
-			solAssert(interfacePredicate && interfacePredicate->isInterface(), "");
-		}
-		/// The children are unordered, so we need to check which is the summary and
-		/// which is the interface.
-
-		Predicate const* summaryPredicate = Predicate::predicate(_graph.nodes.at(summaryId).name);
-		solAssert(summaryPredicate && summaryPredicate->isSummary(), "");
-		/// At this point property 2 from the function description is verified for this node.
-		vector<smtutil::Expression> summaryArgs = _graph.nodes.at(summaryId).arguments;
+	auto callGraph = summaryCalls(_graph, *rootId);
 
-		FunctionDefinition const* calledFun = summaryPredicate->programFunction();
-		ContractDefinition const* calledContract = summaryPredicate->programContract();
+	bool first = true;
+	for (auto summary: callGraph.at(*rootId))
+	{
+		Predicate const* summaryPredicate = Predicate::predicate(summary->name);
+		solAssert(summaryPredicate->isSummary(), "");
```

Comment:
A bit paranoid, isn't it :-)?

Reply:
Do you trust our BFS?

Reply:
I trust the ``if`` before the push in it :-).